### PR TITLE
Consolidate `test_multidynamic_link` and `test_multiply_defined_libsymbols`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Since these tests were largely redundant, consolidate them into a single parameterized test. Specifically, `test_multiply_defined_libsymbols` was more comprehensive but `test_multidynamic_link tested` versioned suffixes and full path linking. `test_multiply_defined_libsymbols` now covers those cases.

### CODE REVIEW

**Comment by sbc100** [comment]
I suspected these two tests were redundant, so I had gemini analyze them and got these results.

I've confirmed that test_multidynamic_link and test_multiply_defined_libsymbols were largely redundant. Both were testing that Emscripten's "fake dylib" model (where .so files are treated
  as object files) correctly handles multiple inclusions of the same library without causing multiple definition errors.

  Comparison
   * test_multiply_defined_libsymbols: A more comprehensive "diamond dependency" test (main -> liba2 -> libA, main -> libb2 -> libA, and main -> libA directly). It also explicitly verifies
     that emcc issues a warning when ignoring a dynamic library during intermediate object file generation.
   * test_multidynamic_link: A simpler "chain dependency" test (main -> libother -> libfile). Its only unique contributions were testing versioned suffixes (e.g., .so.1.2.3) and linking by
     full path rather than -l.

  Action Taken
  I have consolidated the tests by:
   1. Enhancing test_multiply_defined_libsymbols: Added parameterization to cover versioned suffixes and full path linking, ensuring no loss of coverage.
   2. Removing test_multidynamic_link: Since its core logic and unique edge cases are now covered by the more robust test_multiply_defined_libsymbols, it has been removed.

File: test/test_other.py [hunk 1]
```diff
@@ -1342,7 +1342,11 @@ def test_symlink_has_bad_suffix(self):
...
 
-  def test_multiply_defined_libsymbols(self):
+  @parameterized({
+    '': (['-lA'], 'libA.so'),
+    'suffixed': (['./libA.so.1.2.3'], 'libA.so.1.2.3'),
+  })
+  def test_multiply_defined_libsymbols(self, link_flags, so_name):
     create_file('libA.c', 'int mult() { return 1; }')
     create_file('a2.c', 'void x() {}')
     create_file('b2.c', 'void y() {}')
```
Comment (kripken) [inline test/test_other.py]: The other test runs code in these files (printing stuff). Perhaps that is worth adding here?

Comment (sbc100) [reply test/test_other.py]: Done

**Review by dschuff — APPROVED** [review]
